pvl/encoder.py

```diff
@@ -476,18 +510,22 @@ def is_scalar(value) -> bool:
 
         For Python, these correspond to the following:
 
-        * numeric_value: int, float, and Units() whose value
+        * numeric_value: int, float, and Quantity whose value
           is int or float
         * date_time_string: datetime objects
         * text_string_value: str
         * symbol_value: str
 
         """
-        if isinstance(value, Units):
-            if isinstance(value.value, (int, float)):
-                return True
-        elif isinstance(value, (int, float, datetime.date, datetime.datetime,
-                                datetime.time, str)):
+        for quant in self.quantities:
```

Comment:
is there only ever 1 quantity in quantities? If you need to perform the test for multiple quantities this will break the loop at the first return.

Reply:
No, self.quantities is a list of all the quantity classes that you want the encoder to be able to handle.  I want to be able to run through them all, but as soon as I get a match, I do want to break the loop.  That's intended.

Let's say that we have a dict-like that contains an astropy.units.Quantity object that we want to encode.  Let's also assume that we have both astropy and pint installed on our system: if so, when the encoder is instantiated, it will have three elements in its self.quantities list (pvl.collections.Quantity, astropy.units.Quantity, and pint.Quantity), and it is ready to handle any of them.

However, since any given object can only be one of these types of class, if it matches (and it is an int or float, but that's secondary), I want it to return True.  I don't need to test if it is a pint.Quantity if I already know it is an astropy.units.Quantity.

Reply:
okay gotcha